### PR TITLE
Magical shenanigans

### PR DESCRIPTION
**Pull request by jdevnull**
1. Spider totem penalty for outdoor magics should be +2, per MitS pg 157.
2. Add dice from spell defense foci to invisibility resistance tests. If spell pool allocated to spell defense / reflect apply, then an activated spell defense foci should also apply.
3. Fix the logic to enable dropping conjuring libraries at power sites. Casting buffs at a power site doesn't result in an unusual increase in successes, so conjuring there should be fine.
4. Remove auto-extraction of elementals/spirits when using their last service. Instead of adding exceptions for all the sustained nature spirit powers, easier to just require manual dismissal once services have been expended.

This addresses the following:
https://discord.com/channels/564618629467996170/788953927269613608/1241234594850930830
https://discord.com/channels/564618629467996170/1224099384527032431
https://discord.com/channels/564618629467996170/1241603104659603520

### CODE REVIEW

File: src/newmagic.cpp [hunk 1]
```diff
@@ -249,7 +249,7 @@ void totem_bonus(struct char_data *ch, int action, int type, int &target, int &s
   } else if (GET_TOTEM(ch) == TOTEM_SCORPION && (time_info.hours > 6 && time_info.hours < 19)) {
     target += 2;
   } else if (GET_TOTEM(ch) == TOTEM_SPIDER && OUTSIDE(ch)) {
-    target++;
+    target += 2;
```
[the remark this answers is not in the document]
Comment (jdevnull) [reply src/newmagic.cpp:252]: Ah, got it. Will turn this into a comment instead.

File: src/newmagic.cpp [hunk 2]
```diff
@@ -1428,6 +1428,7 @@ int resist_spell(struct char_data *ch, int spell, int force, int sub)
                    GET_OBJ_NAME(eq),
                    GET_OBJ_VNUM(eq),
                    GET_FOCUS_FORCE(eq));
+          break;
```
[the remark this answers is not in the document]
Comment (jdevnull) [reply src/newmagic.cpp:1431]: RAW does not limit the number of any single type of focus (i.e., you can use multiple power foci, multiple spell foci, etc). I believe this was house ruled in awake in order to make the force limits for individual foci meaningful.

[the remark this answers is not in the document]
Comment (jdevnull) [reply src/newmagic.cpp:1431]: I'll change this to use the single highest found, so it works the same way as spell category and specific spell foci.

**Comment by jdevnull** [comment]
5. Allow heal spells to be cast more than once, so long as the target is still mortally wounded. This house rule would be similar to part of how biotech first aid was house ruled and allows the heal spell to be reasonable for player docwagon rescues.

Addresses: https://discord.com/channels/564618629467996170/788953927269613608/1244026373744496661

**Comment by luciensadi** [comment]
LGTM, thanks!

**Pull request merged**